crates/rpc/src/jsonrpc/websocket/logic.rs

```diff
@@ -522,8 +523,9 @@ async fn transaction_status_subscription(
         None = 0,
         Received = 1,
         AcceptedOnL2 = 2,
-        AcceptedOnL1 = 3,
-        Rejected = 4,
+        Reverted = 3,
+        AcceptedOnL1 = 4,
+        Rejected = 5,
```

Comment:
Unfortunately this won't be enough. We'll need two separate fields to properly represent the transactions status. We need both the finality and execution status.

As an example a transaction can be reverted (execution status), but has not yet been accepted by L1 (finality status).

In Rust terms, this would look like this

```rust
enum ExecutionStatus {
  Success,
  Reverted,
}

/// Combined execution and finality status.
enum Status {
  Received,
  Rejected,
  AcceptedOnL2(ExecutionStatus),
  AcceptedOnL1(ExecutionStatus),
}
```